### PR TITLE
Solved issue 24, 25, 26: Set up postgres database connection and implemented functions to modify database for orders page

### PR DESCRIPTION
**Pull request by ziyielenachen**
Solved issue 24, 25, 26 by setting up postgres database connection and implementing functions in the backend that modify the database called “orders” created using the local configuration file. 

#24. Set up the postgres database connection in order_query.js using a configuration file stored locally. Our express backend is connected to the postgresql server, specifically to the database called “orders” so that we can manipulate data-table “order_table” within the database for future purposes.

#25. Created a new table called “order_table” within database “orders”. The table has three columns: {order_number, item, picked_up}. Column “order_number” stores order numbers pulled from Ethical GoogleSheet in data type TEXT; column item stores the name of the item in data type TEXT; and column “picked_up” stores BOOLEAN data to indicate whether that order/row has been picked up by the customer. 
Command used to create such table using postgres:
CREATE TABLE order_table (order_number TEXT, 
item TEXT, picked_up BOOLEAN);

#26. Inside order_query.js, created functions for adding orders to order_table from google sheet, updating the column “picked_up,” and checking order_table against the entries in google sheet. Inside app.js, we utilized these functions by importing them from order_query.js. Specifically, we used the function “addOrders” to add new rows of data to order_table in the database from the google sheet if that order_number&item combination does not already exist in the database. We have not used function “updatePickUp” but will use it when we implement the checkboxes because that is when one will be able to update the pickup status of an order. Moreover, we used function “checkAgainst” to make sure that if the  order has been picked up, meaning that in order_table the row with that specific order and item now has value BOOLEAN true for column “picked_up”, then we update our rows constructed from google sheet’s data before sending over to front end to ensure that the right status is reflected.

### CODE REVIEW

File: backend/orderquery.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+const updatePickUp = (request, response) => {
+  const { order_num, item, picked_up } = request.body;
+  pool.query(
+    "UPDATE order_table SET picked_up = NOT $1 WHERE order_num = $2 AND item = $3",
+    [picked_up, order_num, item],
+    (error, results) => {
+      if (error) {
+        throw error;
+      }
+      //response.status(200).send(`User modified with ID: ${id}`);
```
Comment (abaruah117) [inline backend/orderquery.js]: remove comment

File: backend/app.js [hunk 2]
```diff
@@ -3,8 +3,18 @@ const app = express();
...
+    extended: true
+  })
+);
 
-app.get("/orders", function(req, res) {
+var fs = require("fs");
+app.get("/orders", function (req, res) {
   // Authorization
   fs.readFile("credentials.json", (err, content) => {
     if (err) return console.log("Error loading client secret file:", err);
```
Comment (abaruah117) [inline backend/app.js]: use res.error instead? or console.log if that doesn't work, but don't return console.log

File: backend/app.js [hunk 3]
```diff
@@ -22,11 +32,26 @@ app.get("/orders", function(req, res) {
     sheets.spreadsheets.values.get(
       {
         spreadsheetId: "1ZNnltFhQluexcZrwWpuaIlMLtDQTeAI2WsoRHWgmELg",
-        range: "A2:G"
+        range: "A2:H"
       },
-      (err, response) => {
+      async (err, response) => {
         if (err) return console.log("The API returned an error: " + err);
```
Comment (abaruah117) [inline backend/app.js]: same as above don't return

File: backend/orderquery.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    console.error(error);
+  }
+}
+
+async function rowCount() {
+  const result = await pool.query("SELECT COUNT(*) FROM order_table");
+  return result.rows[0]["count"];
+}
+
+async function newRowCount() {
```
Comment (abaruah117) [inline backend/orderquery.js]: why do you have a duplicate function

File: backend/orderquery.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+const addOrder = (gsOrder, gsItem, gsPickUp) => {
+  pool.query(
+    "INSERT INTO order_table ( order_number, item, picked_up ) SELECT CAST( $1 AS TEXT) AS order_number, $2 AS item, $3 AS picked_up WHERE (NOT EXISTS( SELECT order_number, item FROM order_table WHERE order_number = $1 AND item = $2 ))",
+    [gsOrder, gsItem, gsPickUp],
+    (error, results) => {
+      if (error) {
+        throw error;
+      }
+      //response.status(201).send(`User added with ID: ${results.insertId}`);
```
Comment (abaruah117) [inline backend/orderquery.js]: remove comment

File: backend/app.js [hunk 6]
```diff
@@ -3,8 +3,18 @@ const app = express();
...
+const pg = require("pg");
+app.use(bodyParser.json());
+app.use(
+  bodyParser.urlencoded({
+    extended: true
+  })
+);
 
-app.get("/orders", function(req, res) {
+var fs = require("fs");
```
Comment (trev4ev) [inline backend/app.js]: move this above `app.use` so all imports are together

File: backend/app.js [hunk 7]
```diff
@@ -53,7 +78,6 @@ app.listen(port, () => console.log(`Example app listening on port ${port}!`));
  * @param {getEventsCallback} callback The callback for the authorized client.
  */
 
-const fs = require("fs");
 const readline = require("readline");
```
Comment (trev4ev) [inline backend/app.js]: move to top for imports

**Review by abaruah117 — CHANGES_REQUESTED** [review]
small changes!

File: backend/orderquery.js (hunk 1 above)
Comment (trev4ev) [inline backend/orderquery.js]: why is this commented? looks like something that would be good to have

File: backend/test.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+const express = require("express");
```
Comment (trev4ev) [inline backend/test.js]: Is this file used at all?

File: backend/app.js [hunk 9]
```diff
@@ -3,8 +3,18 @@ const app = express();
 const port = 3001;
 const cors = require("cors");
 app.use(cors());
+const bodyParser = require("body-parser");
+const db = require("./orderquery");
```
Comment (trev4ev) [inline backend/app.js]: change variable name to be order specific

File: backend/orderquery.js [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+    (error, results) => {
+      if (error) {
+        throw error;
+      }
+      //response.status(200).send(`User modified with ID: ${id}`);
+    }
+  );
+};
+
+const getOrders = (request, response) => {
```
Comment (trev4ev) [inline backend/orderquery.js]: don't think we need anymore

**Review by trev4ev — CHANGES_REQUESTED** [review]
Looks good, mainly just refactoring and removing things we don't need

**Comment by trev4ev** [comment]
I appreciate this PR description

File: backend/orderquery.js (hunk 1 above)
Comment (abaruah117) [reply backend/orderquery.js]: +1

File: backend/app.js [hunk 11]
```diff
@@ -2,12 +2,23 @@ const express = require("express");
 const app = express();
 const port = 3001;
 const cors = require("cors");
+const readline = require("readline");
+const { google } = require("googleapis");
+var fs = require("fs");
 app.use(cors());
```
Comment (trev4ev) [inline backend/app.js]: sorry please put this under all the imports LOL

**Review by trev4ev — CHANGES_REQUESTED** [review]
left 1 comment

**Review by trev4ev — APPROVED** [review]

**Review by abaruah117 — APPROVED** [review]
Y e e t

**Pull request merged**